hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/connector/ReadFilesP.java

```diff
@@ -103,13 +141,21 @@ public static ProcessorSupplier supplier(@Nonnull String directory, @Nonnull Str
         return new ProcessorSupplier() {
             static final long serialVersionUID = 1L;
 
+            private transient List<ReadFilesP> processors;
+
             @Nonnull
             @Override
             public Collection<? extends Processor> get(int count) {
                 Charset charsetObj = Charset.forName(charset);
-                return IntStream.range(0, count)
+                processors = IntStream.range(0, count)
                         .mapToObj(i -> new ReadFilesP(directory, charsetObj, glob, count, i))
                         .collect(Collectors.toList());
+                return processors;
+            }
+
+            @Override
+            public void complete(Throwable error) {
+                processors.forEach(p -> uncheckRun(p::close));
```

Comment:
I think we have this pattern in quite a few places. Maybe we should have a utility like CloseableProcessorSupplier?